medium-tests/src/test/java/mediumtest/PathMatchingMediumTests.java

```diff
@@ -0,0 +1,147 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.config.binding.BindingConfigurationDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.config.scope.ConfigurationScopeDto;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.config.scope.DidAddConfigurationScopesParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.file.GetPathTranslationParams;
+import org.sonarsource.sonarlint.core.rpc.protocol.backend.file.GetPathTranslationResponse;
+import org.sonarsource.sonarlint.core.rpc.protocol.common.Language;
+
+import static java.util.concurrent.TimeUnit.SECONDS;
+import static mediumtest.fixtures.ServerFixture.newSonarQubeServer;
+import static mediumtest.fixtures.SonarLintBackendFixture.newBackend;
+import static mediumtest.fixtures.SonarLintBackendFixture.newFakeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.awaitility.Awaitility.waitAtMost;
+
+class PathMatchingMediumTests {
+
+  @Test
+  void it_should_return_no_prefixes_when_matching_did_not_happen() {
+    var server = newSonarQubeServer("10.3")
+      .withProject("projectKey", project ->
+        project.withBranch("main"))
+      .start();
+    backend = newBackend()
+      .withEnabledLanguageInStandaloneMode(Language.JAVA)
+      .withSonarQubeConnection("connectionId", server)
+      .build();
+
+    addConfigurationScope("configScopeId", "connectionId", "projectKey");
+
+    await().during(Duration.ofMillis(300)).untilAsserted(() -> assertThat(getPathTranslation("configScopeId"))
+      .extracting(GetPathTranslationResponse::getIdePathPrefix, GetPathTranslationResponse::getServerPathPrefix)
+      .containsExactly(null, null));
+  }
+
+  @Test
+  void it_should_match_without_prefixes_when_no_local_and_main_branch_server_files_exist() {
+    var server = newSonarQubeServer("10.3")
+      .withProject("projectKey", project ->
+        project.withBranch("main"))
+      .start();
+    backend = newBackend()
+      .withEnabledLanguageInStandaloneMode(Language.JAVA)
+      .withSonarQubeConnection("connectionId", server)
+      .withFullSynchronization()
+      .build();
+
+    addConfigurationScope("configScopeId", "connectionId", "projectKey");
+
+    waitAtMost(3, SECONDS).untilAsserted(() -> assertThat(getPathTranslation("configScopeId"))
+      .extracting(GetPathTranslationResponse::getIdePathPrefix, GetPathTranslationResponse::getServerPathPrefix)
+      .containsExactly("", ""));
+  }
+
+  @Test
+  void it_should_match_without_prefixes_when_local_and_main_branch_server_paths_are_the_same() {
+    var server = newSonarQubeServer("10.3")
+      .withProject("projectKey", project ->
+        project.withBranch("main")
+        .withFile("relative/path/to/a/file"))
+      .start();
+    var client = newFakeClient()
+      .withFile("configScopeId", "relative/path/to/a/file")
```

Comment:
As discussed with Vojtech, I think we could stop coding manually the "stubbing" of the FakeClient. You can directly use Mockito. This allows much more flexibility, and I don't think it is hurting readability. Something like:
```java
var client = spy(newFakeClient().build());
when(client.listAllFilePaths(any()).thenReturn(xxx);
```

Reply:
I was wondering about the same thing. Sometimes using Mockito could be simpler, but in the fake client there are some reasonable defaults, e.g. for branch resolution we return the main branch by default, which is used in several scenarios. There might be other examples. We could give it a try, but in a separate PR